### PR TITLE
flexbox 2020: Implement `align-items` and `align-self`, excluding `baseline`

### PR DESCRIPTION
**Pull request by Manishearth**
This PR is based on https://github.com/servo/servo/pull/27334 for ease of testing

`baseline` is tricky and involves keeping track of a "baseline set"

https://drafts.csswg.org/css-flexbox/#flex-baselines

Part of https://github.com/servo/servo/issues/26639

### CODE REVIEW

File: components/layout_2020/flexbox/layout.rs [hunk 1]
```diff
@@ -1149,14 +1152,29 @@ impl FlexItem<'_> {
     }
 
     /// Return the coordinate of the cross-start side of the content area
-    fn align_along_cross_axis(&self, margin: &FlexRelativeSides<Length>) -> Length {
+    fn align_along_cross_axis(
+        &self,
+        margin: &FlexRelativeSides<Length>,
+        size: &Length,
```
Comment (pcwalton) [inline components/layout_2020/flexbox/layout.rs:1158]: Maybe call this `content_size` or something?

**Comment by pcwalton** [comment]
Looks good other than that one nit

**Comment by jdm** [comment]
```
./tests/html/flexbox-align-self.html:41: no newline at EOF
```

**Comment by Manishearth** [comment]
@bors r=pcwalton p=1

prioritize over  #27334

**Comment by Manishearth** [comment]
@SimonSapin would still appreciate a nonblocking review from you whenever you have time though

**Comment by Manishearth** [comment]
@bors-servo r=pcwalton p=1

oops

**Comment by jdm** [comment]
```
error[E0425]: cannot find value `size` in this scope
    --> components/layout_2020/flexbox/layout.rs:1168:49
     |
1168 |                         let margin_box_cross = *size + self.pbm_auto_is_zero.cross;
     |                                                 ^^^^ not found in this scope
error[E0425]: cannot find value `size` in this scope
    --> components/layout_2020/flexbox/layout.rs:1172:49
     |
1172 |                         let margin_box_cross = *size + self.pbm_auto_is_zero.cross;
     |                                                 ^^^^ not found in this scope
```

**Comment by Manishearth** [comment]
```
test test_hang_monitoring_unregister ... ok(B
error: test failed, to rerun pass '-p background_hang_monitor --test hang_monitor_tests'
Caused by:
  process didn't exit successfully: `/repo/target/debug/deps/hang_monitor_tests-2180465ee33d06ac` (signal: 27)
```

@bors-servo retry

 - https://github.com/servo/servo/issues/27315

**Comment by Manishearth** [comment]
@bors-servo retry

WR build scripts?

**Comment by CYBAI** [comment]
that's #27039 👀

**Comment by Manishearth** [comment]
```
     7: _ZN12_GLOBAL__N_126builtin_variable_generator11add_uniformEPK9glsl_typePKc
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/glsl-optimizer/src/compiler/glsl/builtin_variables.cpp:449
     8: _ZN12_GLOBAL__N_126builtin_variable_generator17generate_uniformsEv
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/glsl-optimizer/src/compiler/glsl/builtin_variables.cpp:983
     9: _Z31_mesa_glsl_initialize_variablesP9exec_listP22_mesa_glsl_parse_state
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/glsl-optimizer/src/compiler/glsl/builtin_variables.cpp:1553
    10: _Z16_mesa_ast_to_hirP9exec_listP22_mesa_glsl_parse_state
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/glsl-optimizer/src/compiler/glsl/ast_to_hir.cpp:131
    11: glslopt_optimize
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/glsl-optimizer/src/compiler/glsl/glsl_optimizer.cpp:673
    12: glslopt::Context::optimize
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/glslopt-0.1.2/src/lib.rs:48
    13: build_script_build::write_optimized_shaders::{{closure}}
               at /root/.cargo/git/checkouts/webrender-479e138e1c1b9e8b/34d968a/webrender/build.rs:159
    14: core::ops::function::impls::<impl core::ops::function::Fn<A> for &F>::call
               at /rustc/2753fab7ce3647033146b07c8b6c9f4856a910b0/src/libcore/ops/function.rs:243
    15: build_parallel::compile_object
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/build-parallel-0.1.1/src/lib.rs:24
    16: build_parallel::compile_objects::{{closure}}::{{closure}}
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/build-parallel-0.1.1/src/lib.rs:90
    17: crossbeam_utils::thread::ScopedThreadBuilder::spawn::{{closure}}
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.7.2/src/thread.rs:415
    18: crossbeam_utils::thread::ScopedThreadBuilder::spawn::{{closure}}
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.7.2/src/thread.rs:423
    19: <alloc::boxed::Box<F> as core::ops::function::FnMut<A>>::call_mut
               at /rustc/2753fab7ce3647033146b07c8b6c9f4856a910b0/src/liballoc/boxed.rs:1088
    20: crossbeam_utils::thread::ScopedThreadBuilder::spawn::{{closure}}
               at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-utils-0.7.2/src/thread.rs:431
    21: std::sys_common::backtrace::__rust_begin_short_backtrace
               at /rustc/2753fab7ce3647033146b07c8b6c9f4856a910b0/src/libstd/sys_common/backtrace.rs:130
    22: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
               at /rustc/2753fab7ce3647033146b07c8b6c9f4856a910b0/src/libstd/thread/mod.rs:475
    23: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
```

**Comment by SimonSapin** [comment]
Looks good, thanks!

**Pull request merged**